cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/SwitchesChain.java

```diff
@@ -0,0 +1,262 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.powsybl.cgmes.conversion;
+
+import java.util.*;
+
+import com.powsybl.iidm.network.Bus;
+import com.powsybl.iidm.network.Connectable;
+import com.powsybl.iidm.network.IdentifiableType;
+import com.powsybl.iidm.network.Switch;
+import com.powsybl.iidm.network.Terminal;
+import com.powsybl.iidm.network.VoltageLevel;
+
+/**
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
+ */
+class SwitchesChain {
+    private final NodeConnectables nodeConnectables;
+    private final Adjacency adjacency;
+    private final VoltageLevel vl;
+    private final Switch sw;
+    private final boolean isNodeBreakerModel;
+
+    SwitchesChain(VoltageLevel vl, Switch sw, boolean isNodeBreakerModel) {
+        this.vl = vl;
+        this.sw = sw;
+        this.isNodeBreakerModel = isNodeBreakerModel;
+        nodeConnectables = new NodeConnectables(vl, isNodeBreakerModel);
+        adjacency = new Adjacency(vl, sw.getId(), isNodeBreakerModel);
```

Comment:
With the new classes `NodeConnectables` and `Adjacency` you're just reconstructing the graph don't you? I don't see which specific need forced you to do that instead of using the existing topology traversers, maybe I missed something. If the existing topology traversers miss something you need, we could also try to improve them, rather than adding this kind of code here.

Reply:
I used the existing traverse for `nodeBreaker `and I implemented a new one for `busBreaker`. I added this new traverse method in the `BusBreakerView `interface of the `VoltageLevel `class. Should I label the PR as Breaking Change?